src/tcti-tabrmd.c

```diff
@@ -612,46 +513,135 @@ tabrmd_conf_parse (char *conf_str,
+    if (fd == -1) {
+        g_critical ("unable to get receive handle from GUnixFDList: %s",
+                    error->message);
+        rc = TSS2_TCTI_RC_GENERAL_FAILURE;
+        goto out;
+    }
+    sock = g_socket_new_from_fd (fd, NULL);
+    TSS2_TCTI_TABRMD_SOCK_CONNECT (context) = \
+        g_socket_connection_factory_create_connection (sock);
+    TSS2_TCTI_TABRMD_ID (context) = id;
+out:
+    g_clear_pointer (&fds_variant, g_variant_unref);
+    g_clear_error (&error);
+    g_clear_object (&sock);
+    g_clear_object (&fd_list);
+    return rc;
+}
 /*
  * The longest configuration string we'll take. Each dbus name can be 255
  * characters long (see dbus spec). The bus_types that we support are
  * 'system' or 'session' (255 + 7 = 262). 'bus_type=' and 'bus_name=' are
  * each another 9 characters for a total of 280.
  */
 #define CONF_STRING_MAX 280
-static TSS2_RC
+TSS2_RC
 Tss2_Tcti_Tabrmd_Init (TSS2_TCTI_CONTEXT *context,
                        size_t            *size,
                        const char        *conf)
 {
-    TSS2_RC ret;
+    GError *error = NULL;
     size_t conf_len;
-    char *conf_copy;
-    tabrmd_conf_t tabrmd_conf = {
-        TCTI_TABRMD_DBUS_TYPE_DEFAULT,
-        TCTI_TABRMD_DBUS_NAME_DEFAULT,
-    };
+    char *conf_copy = NULL;
+    TSS2_RC rc;
+    tabrmd_conf_t tabrmd_conf = TABRMD_CONF_INIT_DEFAULT;
 
+    if (context == NULL && size != NULL) {
+        *size = sizeof (TSS2_TCTI_TABRMD_CONTEXT);
+        return TSS2_RC_SUCCESS;
+    }
+    if (size == NULL) {
+        return TSS2_TCTI_RC_BAD_VALUE;
+    }
     if (conf != NULL) {
         conf_len = strlen (conf);
         if (conf_len > CONF_STRING_MAX) {
             return TSS2_TCTI_RC_BAD_VALUE;
         }
-        conf_copy = strdup (conf);
+        conf_copy = g_strdup (conf);
         if (conf_copy == NULL) {
             g_critical ("Failed to duplicate config string: %s", strerror (errno));
             return TSS2_TCTI_RC_GENERAL_FAILURE;
         }
 
-        ret = tabrmd_conf_parse (conf_copy, &tabrmd_conf);
-        if (ret != TSS2_RC_SUCCESS)
-            return ret;
+        rc = tabrmd_conf_parse (conf_copy, &tabrmd_conf);
+        if (rc != TSS2_RC_SUCCESS) {
+            goto out;
+        }
+    }
+    /* Register dbus error mapping for tabrmd. Gets us RCs from Gerror codes */
+    TABRMD_ERROR;
+    init_tcti_data (context);
+    TSS2_TCTI_TABRMD_PROXY (context) =
+        tcti_tabrmd_proxy_new_for_bus_sync (tabrmd_conf.bus_type,
+                                            G_DBUS_PROXY_FLAGS_NONE,
+                                            tabrmd_conf.bus_name,
+                                            TABRMD_DBUS_PATH,
+                                            NULL,
+                                            &error);
+    if (TSS2_TCTI_TABRMD_PROXY (context) == NULL) {
+        g_critical ("failed to allocate dbus proxy object: %s", error->message);
+        rc = TSS2_TCTI_RC_NO_CONNECTION;
```

Comment:
There is something not right here. rc is assigned an error code, which is then ignored and TSS2_RC_SUCCESS is returned.